mono/mini/mini-exceptions.c

```diff
@@ -1223,7 +1225,10 @@ mono_handle_exception_internal_first_pass (MonoContext *ctx, MonoObject *obj, gi
 	mono_ex = (MonoException*)obj;
 	initial_trace_ips = mono_ex->trace_ips;
 
-	if (mono_object_isinst (obj, mono_defaults.exception_class)) {
+	isinst = mono_object_isinst_checked (obj, mono_defaults.exception_class, &error) == obj;
+	mono_error_raise_exception (&error); /* FIXME don't raise here */
```

Comment:
We should assert here since this is running within the exception unwinder and things will go freaky if we throw from here.